presentation-compiler/src/main/scala/org/alephium/ralph/lsp/pc/search/gotodef/soft/GoToDefIdentifier.scala

```diff
@@ -307,21 +307,21 @@ private object GoToDefIdentifier extends StrictImplicitLogging {
       from = from,
       anchor = target.data.index
     ) {
-      case Node(variable: SoftAST.VariableDeclaration, _) if !detectCallSyntax || !target.isReferenceCall() =>
+      case Node(variable: SoftAST.VariableDeclaration, _) if !detectCallSyntax || (!target.isReferenceCall() && !target.isWithinEmit()) =>
         searchExpression(
           expression = variable,
           target = target,
           sourceCode = sourceCode
         )
 
-      case Node(assignment: SoftAST.TypeAssignment, _) if !detectCallSyntax || !target.isReferenceCall() =>
+      case Node(assignment: SoftAST.TypeAssignment, _) if !detectCallSyntax || (!target.isReferenceCall() && !target.isWithinEmit()) =>
         searchExpression(
           expression = assignment,
           target = target,
           sourceCode = sourceCode
         )
 
-      case Node(binding: SoftAST.MutableBinding, _) if !detectCallSyntax || !target.isReferenceCall() =>
+      case Node(binding: SoftAST.MutableBinding, _) if !detectCallSyntax || (!target.isReferenceCall() && !target.isWithinEmit()) =>
         searchExpression(
```

Comment:
would it make sense to extract those condition in a function? as it's 3 time the same.

Your call

Reply:
Good point 👍🏼. I'll save this change for later. Implementing go-to-def for others types might require additional conditions for one over the others.